main/influxdb_watcher.go

```diff
@@ -0,0 +1,205 @@
+		Database:  broker.MonascaDB,
+		Precision: "s",
+	})
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+
+	// Create a point and add to batch
+	tags := map[string]string{"watcher": "influxdb"}
+	fields := map[string]interface{}{
+		"uuid":      message.UUID,
+		"startTime": message.SentTime,
+	}
+
+	pt, err := client.NewPoint("watcher.influxdb", tags, fields, time.Now())
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+
+	bp.AddPoint(pt)
+
+	// Write the batch
+	if err := broker.Connection.Write(bp); err != nil {
+		log.Fatal(err)
+		return err
+	}
+	return nil
+}
+
+// ReadMessage query InfluxDB for latest measurement
+func (broker *InfluxdbBroker) ReadMessage(timeout time.Duration) (*[]byte, error) {
+	cmd := fmt.Sprintf("SELECT last(startTime), last(uuid) FROM \"%s\"", "watcher.influxdb")
+	res, err := queryDB(broker, cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	message := Message{
+		UUID:     res[0].Series[0].Values[0][2].(string),
+		SentTime: res[0].Series[0].Values[0][1].(string),
+	}
+	bytes, err := json.Marshal(message)
+	if err != nil {
+		panic("json encoding failed")
+	}
+	return &bytes, nil
+}
+
+// queryDB convenience function to query the database
+func queryDB(broker *InfluxdbBroker, cmd string) (res []client.Result, err error) {
+	q := client.Query{
+		Command:  cmd,
+		Database: broker.MonascaDB,
+	}
+	response, err := broker.Connection.Query(q)
+	if err != nil {
+		return nil, err
+	}
+	if response.Error() != nil {
+		return nil, response.Error()
+	}
+	return response.Results, nil
+}
+
+func main() {
+	configuration := watcherConfiguration{}
+	err := configEnv.Parse(&configuration)
+	if err != nil {
+		log.Fatalf("Failed to parse environment variables. %s", err)
+	}
+
+	influxdbAddress := configuration.InfluxdbAddress
+	watcher.ValidateConfString("INFLUXDB_ADDRESS", influxdbAddress)
+	username := configuration.Username
+	watcher.ValidateConfString("INFLUXDB_USERNAME", username)
+	password := configuration.Password
+	watcher.ValidateConfString("INFLUXDB_PASSWORD", password)
+	prometheusEndpoint := configuration.PrometheusEndpoint
+	watcher.ValidateConfString("PROMETHEUS_ENDPOINT", prometheusEndpoint)
+
+	log.Infof("Using InfluxDB Address %s", influxdbAddress)
+
+	influxdbBroker := InfluxdbBroker{MonascaDB: "mon"}
+	period := time.Duration(configuration.Period) * time.Second
+	timeout := time.Duration(configuration.Timeout) * time.Second
+
+	// Create the watcher now so it has NOT_STARTED status while we are initializing
+	// the connections to InfluxDB
+	watcher := watcher.CreateWatcher(&influxdbBroker, period, 1, timeout, "influxdb")
+
+	go func() {
+		// Start prometheus endpoint
+		http.Handle("/metrics", promhttp.Handler())
+		log.Fatal(http.ListenAndServe(prometheusEndpoint, nil))
+	}()
+
+	var connection client.Client
+	connection, err = startConnect(influxdbAddress, username, password)
+	if err != nil {
+		log.Fatalf("Starting connection to InfluxDB failed: %s", err)
```

Comment:
What if influxdb isn't up, yet? Look at the other watchers, they retry in a loop.